client/my-sites/upgrades/domain-management/dns/office-365.jsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { localize } from 'i18n-calypso';
+import FormButton from 'components/forms/form-button';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormFooter from 'my-sites/upgrades/domain-management/components/form-footer';
+import FormInputValidation from 'components/forms/form-input-validation';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+import notices from 'notices';
+import * as upgradesActions from 'lib/upgrades/actions';
+
+class Office365 extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = { token: '', submitting: false };
+	}
+
+	onChange = ( event ) => {
+		const { value } = event.target;
+		this.setState( { token: value } );
+	}
+
+	onAddDnsRecords = ( event ) => {
+		event.preventDefault();
+		this.setState( { submitting: true } );
+		upgradesActions.addDnsOffice( this.props.selectedDomainName, this.state.token, ( error ) => {
+			if ( error ) {
+				notices.error( error.message || this.props.translate( 'The DNS record has not been added.' ) );
+				this.setState( { submitting: false } );
+			} else {
+				notices.success( this.props.translate( 'The DNS records have been added.' ), {
+					duration: 5000
+				} );
+			}
+		} );
+	}
+
+	render() {
+		const isDataValid = this.state.token.match( /^MS=ms\d+$/ );
+
+		return (
+			<form className="dns__office365">
+				<div className="dns__form-content">
+					<FormFieldset>
```

Comment:
Since we are automating nicely for the user which not documented in Office365 or in our support. What do you think about adding a small description of saying we'll set everything up for them if they just give type the verification code?